pkg/metrics/features/metrics.go

```diff
@@ -804,6 +830,34 @@ func NewMetrics(withDefaults bool) Metrics {
 			Subsystem: subsystemNP,
 			Name:      "cilium_clusterwide_envoy_config_present",
 		}),
+
+		ACLBClusterMeshEnabled: metric.NewGaugeVecWithLabels(metric.GaugeOpts{
+			Help:      "Clustermesh mode enabled on the agent",
+			Namespace: metrics.Namespace,
+			Subsystem: subsystemACLB,
+			Name:      "clustermesh_enabled",
+		}, metric.Labels{
+			{
+				Name: "mode", Values: func() metric.Values {
+					if !withDefaults {
+						return nil
+					}
+					return metric.NewValues(
+						defaultClusterMeshMode...,
+					)
+				}(),
+			},
+			{
+				Name: "max_connected_clusters", Values: func() metric.Values {
+					if !withDefaults {
+						return nil
+					}
+					return metric.NewValues(
+						defaultClusterMeshMaxConnectedClusters...,
+					)
+				}(),
+			},
```

Comment:
I wonder if it could make sense to break this into a separate metric. The reason being that the mode is specific to the clusters we are connected to, while the max_connected_clusters number of an agent specific setting (all clusters in a clustermesh must use the same value). WDYT?